### ISSUE TITLE
Cargo check mode is broken: no code checking

### ISSUE THREAD

**Issue by MSxDOS** [intellij-rust/intellij-rust#2925]
## Environment

* **Intellij-Rust plugin version:** 
    0.2.0.2107-182
* **Rust toolchain version:**
    rustc 1.31.0-nightly (96cafc53c 2018-10-09)  
    binary: rustc
    commit-hash: 96cafc53cfc6667a03c8e77d8e0a2fc96555ff6b
    commit-date: 2018-10-09
    host: x86_64-pc-windows-msvc
    release: 1.31.0-nightly
    LLVM version: 8.0
* **IDE name and version:**
    IntelliJ IDEA 2018.2.4 (Community Edition)
    Build #IC-182.4505.22, built on September 18, 2018
    JRE: 1.8.0_152-release-1248-b8 amd64
    VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
    Windows 7 6.1
* **Operating system:**
    Windows 7 x64

## Problem description
The plugin does show code warnings\errors when in cargo check mode. IIRC, it started a few nightlies ago.

## Steps to reproduce
```Rust
pub fn broken_function() -> String {
    use std::io::Write;
    1u32 + 2isize - "literal"
}
```
No warnings or errors from the plugin.
